### DISCUSSION TITLE
choose an alternative route

### DISCUSSION THREAD

**Discussion by BluEe33** [organicmaps/organicmaps#5009] [Q&A]
![微信图片_20230419191115](https://user-images.githubusercontent.com/127585800/233057424-25ac2941-3493-4c26-8120-03601915f85d.jpg)
This is the original bicycle navigation path
![微信图片_20230419191129](https://user-images.githubusercontent.com/127585800/233057563-70fec35a-085a-4d2e-9562-1ec0542c1376.jpg)
This is the original walking path. Obviously, neither of these two path choices is the shortest path. I ran organicmaps on linux, and modified the routing weights of routing_common/bicycle_model.cpp and pedestrian_model.cpp to make them all choose the shortest path. .
![3%HS 3L5_G)M2$AK1KINB](https://user-images.githubusercontent.com/127585800/233057948-10413a1b-7382-4c45-bc1e-ce5d251eceb7.png)
![{T0G1WK1 76HO_``34_`K7P](https://user-images.githubusercontent.com/127585800/233058258-8f16bb87-32f3-4c57-90ee-3b4308418770.png)
But I also want to keep the original navigation path. Is there any solution?

**Comment by biodranik** [comment]
These are the fastest paths. What is your case? What do you want to implement? Do you want to contribute to OM or are you working on your own product?

**Reply by BluEe33** [reply]
I am doing a graduation project with research significance. Of course, it would be best if I can contribute to organicmaps. I want to add a feature that can choose an alternate route, allowing the user to choose the originally designed route or the shortest possible route that I have modified.

**Reply by biodranik** [reply]
We would be grateful if you can implement it properly. Please check for existing PRs and issues about alternate or alternative routes. 

Under the hood, some work should be done to run algorithm in parallel with different rules/weights. 

Then routes should be rendered and made selectable. 

It's a lot of work, and we would be happy to help.
